File: Documentation/5.2/Samples/nodejs/ClientApi/Configuration/LoadBalance/readBalance.js
import { DocumentStore } from "ravendb";

const documentStore = new DocumentStore();

async function readBalance() {
    {
        //region readBalance_1
        // Initialize 'readBalanceBehavior' on the client: 
        // ===============================================
        
        const documentStore = new DocumentStore(["serverUrl_1", "serverUrl_2", "..."], "DefaultDB");

        // For example:
        // With readBalanceBehavior set to: 'FastestNode':
        // Client READ requests will address the fastest node
        // Client WRITE requests will address the preferred node
        documentStore.conventions.readBalanceBehavior = "FastestNode";
        
        documentStore.initialize();
        //endregion    
    }
    {
        //region readBalance_2
        // Setting 'readBalanceBehavior' on the server by sending an operation:
        // ====================================================================
        
        // Define the client configuration to put on the server
        const configurationToSave = {
            // Replace 'FastestNode' (from example above) with 'RoundRobin'
            readBalanceBehavior: "RoundRobin"
        };
        
        // Define the put configuration operation for the DEFAULT database
        const putConfigurationOp = new PutClientConfigurationOperation(configurationToSave));

        // Execute the operation by passing it to maintenance.send
        await documentStore.maintenance.send(putConfigurationOp);

        // After the operation has executed:
        // All WRITE requests will continue to address the preferred node 
        // READ requests, per session, will address a different node based on the RoundRobin logic
        //endregion
    }
    {
        //region readBalance_3
        // Setting 'readBalanceBehavior' on the server by sending an operation:
        // ====================================================================

        // Define the client configuration to put on the server
        const configurationToSave = {
            // Replace 'FastestNode' (from example above) with 'RoundRobin'
            readBalanceBehavior: "RoundRobin"
        };

        // Define the put configuration operation for ALL databases
        const putConfigurationOp = new PutServerWideClientConfigurationOperation(configurationToSave));

        // Execute the operation by passing it to maintenance.server.send
        await documentStore.maintenance.server.send(putConfigurationOp);

        // After the operation has executed:
        // All WRITE requests will continue to address the preferred node 
        // READ requests, per session, will address a different node based on the RoundRobin logic
        //endregion
    }
}

File: Documentation/5.4/Samples/csharp/Raven.Documentation.Samples/ClientApi/Configuration/LoadBalance/ReadBalance.cs
using Raven.Client.Documents;
using Raven.Client.Documents.Conventions;
using Raven.Client.Documents.Operations.Configuration;
using Raven.Client.Http;
using Raven.Client.ServerWide.Operations.Configuration;

namespace Raven.Documentation.Samples.ClientApi.Configuration.LoadBalance
{
    public class ReadBalanceExamples
    {
        public ReadBalanceExamples()
        {
            #region ReadBalance_1
            // Initialize 'ReadBalanceBehavior' on the client:
            var documentStore = new DocumentStore
            {
                Urls = new[] { "ServerURL_1", "ServerURL_2", "..." },
                Database = "DefaultDB",
                Conventions = new DocumentConventions
                {
                    // With ReadBalanceBehavior set to: 'FastestNode':
                    // Client READ requests will address the fastest node
                    // Client WRITE requests will address the preferred node
                    ReadBalanceBehavior = ReadBalanceBehavior.FastestNode
                }
            }.Initialize();
            #endregion
            
            #region ReadBalance_2
            // Setting 'ReadBalanceBehavior' on the server by sending an operation:
            using (documentStore)
            {
                // Define the client configuration to put on the server
                var clientConfiguration = new ClientConfiguration
                {
                    // Replace 'FastestNode' (from example above) with 'RoundRobin'
                    ReadBalanceBehavior = ReadBalanceBehavior.RoundRobin
                };
                
                // Define the put configuration operation for the DEFAULT database
                var putConfigurationOp = new PutClientConfigurationOperation(clientConfiguration);
                
                // Execute the operation by passing it to Maintenance.Send
                documentStore.Maintenance.Send(putConfigurationOp);
                
                // After the operation has executed:
                // All WRITE requests will continue to address the preferred node 
                // READ requests, per session, will address a different node based on the RoundRobin logic
            }
            #endregion
            
            #region ReadBalance_3
            // Setting 'ReadBalanceBehavior' on the server by sending an operation:
            using (documentStore)
            {
                // Define the client configuration to put on the server
                var clientConfiguration = new ClientConfiguration
                {
                    // Replace 'FastestNode' (from example above) with 'RoundRobin'
                    ReadBalanceBehavior = ReadBalanceBehavior.RoundRobin
                };
                
                // Define the put configuration operation for the ALL databases
                var putConfigurationOp = new PutServerWideClientConfigurationOperation(clientConfiguration);
                
                // Execute the operation by passing it to Maintenance.Server.Send
                documentStore.Maintenance.Server.Send(putConfigurationOp);
                
                // After the operation has executed:
                // All WRITE requests will continue to address the preferred node 
                // READ requests, per session, will address a different node based on the RoundRobin logic
            }
            #endregion
        }
    }
}
